pygmt/src/x2sys_cross.py

```diff
@@ -71,7 +71,12 @@ def tempfile_from_dftrack(track, suffix):
     Z="trackvalues",
 )
 @kwargs_to_strings(R="sequence")
-def x2sys_cross(tracks=None, outfile=None, **kwargs):
+def x2sys_cross(
+    tracks=None,
+    output_type: Literal["pandas", "file"] = "pandas",
```

Comment:
Since the only two options are `pandas` or `file`, we probably don't need an `output_type` parameter, and can revert to the previous code where a `pandas.DataFrame` output is returned when `outfile` is not set?

Reply:
Done in 5f045069c76fa316d550bcf7e0cea75149045e5b.

Reply:
Perhaps we should revert the changes in https://github.com/GenericMappingTools/pygmt/pull/3191 since it's not needed by this PR.

Reply:
Hmm, you're right. Sorry, should have realized that earlier :sweat_smile: